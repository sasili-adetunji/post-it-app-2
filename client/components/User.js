import React from 'react';
import { ListItem } from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import API from '../Api';

class User extends React.Component {
  /**
   * Creates an instance of User.
   * @param {any} props
   * @memberof User
   */
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <ListItem
        href={`/#/dashboard/users/${this.props.user.userId}`}
      > {this.props.user.username} </ListItem>
      </div>
    );
  }
}

export default User;
