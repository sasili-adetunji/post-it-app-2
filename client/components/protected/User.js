import React from 'react';

/**
 *
 * @class Message
 * @extends {React.Component}
 */
class Message extends React.Component {
  constructor(props) {
    super(props);
  }

  /**
   *
   * renders the message components
   * @returns {void}
   * @memberof Message
   */

  render() {
    return (
      <div className="side-menu-container">
        {/* <ul className="nav navbar-nav"> */}
          <li><span />
            {this.props.user.userName} </li>
        {/* </ul> */}
      </div>
    );
  }
}
export default Message;
