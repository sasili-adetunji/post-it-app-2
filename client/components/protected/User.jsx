import React from 'react';

/**
 * creates user component
 * @class User
 * @extends {React.Component}
 */
class Message extends React.Component {

  /**
 * Displays the List of Users in a group
 * @param props
 * @export
 * @class User
 * @extends {Component}
 */

  render() {
    return (
      <div className="side-menu-container list-group-item">
        <li><span />
          <b> {this.props.user.userName} </b> </li>
      </div>
    );
  }
}
export default Message;
