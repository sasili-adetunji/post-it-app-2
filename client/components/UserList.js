import React from 'react';
import { CardHeader, CardTitle, Card } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { List } from 'material-ui/List';
import PostItActions from '../actions/PostItActions';
import PostItStore from '../stores/PostItStore';
import User from './User';


class UserList extends React.Component {
  constructor(props) {
    super(props);
  }

  /**
   * renders the userlist components
   * @returns {void}
   * @memberof UserList
   */
  render() {
    const userNodes = this.props.users.map((user, i) => {
      return (
        <User user={user} key={i} />
      );
    });
    return (
      <MuiThemeProvider>
        <Card>
          <List>
            <CardTitle title="User List" />
            {userNodes}
          </List>
        </Card>
      </MuiThemeProvider>
    );
  }
        }


export default UserList;
