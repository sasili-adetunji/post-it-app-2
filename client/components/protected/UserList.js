import React from 'react';
import AddMember from './AddMember';
import User from './User';
import PostItStore from '../../stores/PostItStore';
import PostItActions from '../../actions/PostItActions';


/**
 * creates userlist components
 * @class UserList
 * @extends {React.Component}
 */
class UserList extends React.Component {

  /** renders the message components
   * @returns { void }
   * @memberof UserList
   */

  render() {
    const userNodes = this.props.user.map((user, i) => {
      return (
        <User user={user} key={i} />
      );
    });
    return (
      <div>
        <div>
          <AddMember groupId={this.props.selectedGroup[0]} user={this.props.usernames} />
        </div>
        <div className="headerlist"> <h4> Members </h4> </div>
        {userNodes}
      </div>
    );
  }
}
export default UserList;
