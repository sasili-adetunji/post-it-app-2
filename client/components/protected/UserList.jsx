import React from 'react';
import User from './User';

/**
 * @description Displays a list of users in a group
 *
 * @function UserList
 *
 * @returns {JSX} list of a users in a group
 */

const UserList = ({ users }) => {
  const userNodes = users.map((user, i) => (
    <User user={user} key={i} />
    ));
  return (
    <div className="groupMembers">
      <div className="groupList">
        <h4 className="card-header">Group Members </h4>
        {userNodes}
      </div>
    </div>
  );
};
export default UserList;
