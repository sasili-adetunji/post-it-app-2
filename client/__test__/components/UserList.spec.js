import React from 'react';
import { shallow, render } from 'enzyme';
import UserList from '../../components/protected/UserList.jsx';
import User from '../../components/protected/User.jsx'; 

describe('UserList components', () => {
  const component = shallow(<UserList />);

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain three div', () => {
    expect(component.find('div').length).toEqual(3);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});
