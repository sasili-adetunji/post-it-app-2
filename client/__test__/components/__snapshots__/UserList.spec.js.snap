// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="headerlist"
    >
       
      <h4
        className="card-header"
      >
         Members 
      </h4>
       
    </div>
    <div
      className="userList"
    />
  </div>,
  "nodes": Array [
    <div>
      <div
        className="headerlist"
      >
         
        <h4
          className="card-header"
        >
           Members 
        </h4>
         
      </div>
      <div
        className="userList"
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <UserList />,
    "_instance": UserList {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <div
        className="headerlist"
      >
         
        <h4
          className="card-header"
        >
           Members 
        </h4>
         
      </div>
      <div
        className="userList"
      />
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserList />,
}
`;
